src/interpreter/ArithmeticHelper.ts

```diff
@@ -508,6 +515,24 @@ export class ArithmeticHelper {
       return [0, val]
     }
   }
+
+  public ExtendedNumberFactory(value: number, type: NumberType, format?: FormatInfo): ExtendedNumber {
+    switch (type) {
+      case NumberType.NUMBER_RAW:
+        return value
+      case NumberType.NUMBER_CURRENCY: {
+        return new CurrencyNumber(value, format ?? this.config.currencySymbol[0])
```

Comment:
if the `currencySymbols` are empty, then the correct semantics is `undefined`, right?

Reply:
I see two options here:
- validate array length to be at least > 1, we need currency for PMV function and others
- it is possible to switch off the currency symbol and PMV will return just a number without format, but this should be done with `currencySymbols: undefined` or `null`, because doing it with an empty array `currencySymbols: []` IMO looks odd

Reply:
`currencySymbols: undefined` is explicitly asking for default value, which is '$'.
IMO setting it to [] is perfectly understandable way of doing exactly what you are asking for: I want type hints with currency, but no predefined symbol for currency. `format` accepts `undefined` anyway, so I think this is WAD.